lang/test/org/partiql/lang/eval/EvaluatingCompilerDateTimeTests.kt

```diff
@@ -34,4 +46,197 @@ class EvaluatingCompilerDateTimeTests : EvaluatorTestBase() {
+            case("TIME '23:59:59'", "23:59:59", TimeForTest(23, 59, 59, 0, 0)),
+            case("TIME (9) '12:24:12.123'", "12:24:12.123000000", TimeForTest(12, 24, 12, 123000000, 9)),
+            case("TIME '12:24:12.12300'", "12:24:12.12300", TimeForTest(12, 24, 12, 123000000, 5)),
+            case("TIME (3) '12:24:12.12300'", "12:24:12.123", TimeForTest(12, 24, 12, 123000000, 3)),
+            case("TIME (4) '12:24:12.12300'", "12:24:12.1230", TimeForTest(12, 24, 12, 123000000, 4)),
+            case("TIME (4) '12:24:12.123'", "12:24:12.1230", TimeForTest(12, 24, 12, 123000000, 4)),
+            case("TIME (0) '12:59:59.9'", "13:00:00", TimeForTest(13, 0,0, 0, 0)),
+            case("TIME WITH TIME ZONE '00:00:00'", "00:00:00${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(0,0,0,0,0, LOCAL_TZ_MINUTES)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123'", "12:24:12.12${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(12, 24, 12, 120000000, 2, LOCAL_TZ_MINUTES)),
+            case("TIME WITH TIME ZONE '12:24:12.12300'", "12:24:12.12300${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(12, 24, 12, 123000000, 5, LOCAL_TZ_MINUTES)),
+            case("TIME (3) WITH TIME ZONE '12:24:12.12300'", "12:24:12.123${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(12, 24, 12, 123000000, 3, LOCAL_TZ_MINUTES)),
+            case("TIME (4) WITH TIME ZONE '12:24:12.12300'", "12:24:12.1230${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(12, 24, 12, 123000000, 4, LOCAL_TZ_MINUTES)),
+            case("TIME (4) WITH TIME ZONE '12:24:12.123'", "12:24:12.1230${LOCAL_TIMEZONE_OFFSET.getOffsetHHmm()}", TimeForTest(12, 24, 12, 123000000, 4, LOCAL_TZ_MINUTES)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123-00:00'", "12:24:12.12+00:00", TimeForTest(12, 24, 12, 120000000, 2, 0)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123+00:00'", "12:24:12.12+00:00", TimeForTest(12, 24, 12, 120000000, 2, 0)),
+            case("TIME (2) WITH TIME ZONE '12:24:12.123+05:30'", "12:24:12.12+05:30", TimeForTest(12, 24, 12, 120000000, 2, 330)),
+            case("TIME (5) WITH TIME ZONE '12:24:12.123678+05:30'", "12:24:12.12368+05:30", TimeForTest(12, 24, 12, 123680000, 5, 330)),
+            case("TIME (2) WITH TIME ZONE '12:59:59.135-05:30'", "12:59:59.14-05:30", TimeForTest(12, 59, 59, 140000000, 2, -330)),
+            case("TIME (2) WITH TIME ZONE '12:59:59.134-05:30'", "12:59:59.13-05:30", TimeForTest(12, 59, 59, 130000000, 2, -330)),
+            case("TIME '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) '01:01:12' IS TIME", "true"),
+            case("TIME WITH TIME ZONE '12:25:12.123456' IS TIME", "true"),
+            case("TIME (2) WITH TIME ZONE '01:01:12' IS TIME", "true"),
+            case("'01:01:12' IS TIME", "false")
+        )
+    }
+
+    private val RANDOM_GENERATOR = generateRandomSeed()
+
+    private fun generateRandomSeed() : Random {
+        val seed = Random(1).nextInt()
```

Comment:
I don't think you intended to include `1` here since that will effectively be using the same seed with every run of the tests, thus, you won't get the best coverage.

Reply:
I see. Good point. I thought the generated seed was different every time. I was wrong. Changed it. Thanks.